extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/LocalHostOnlyFilter.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.devui.runtime;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.jboss.logging.Logger;
+
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.RoutingContext;
+
+public class LocalHostOnlyFilter implements Handler<RoutingContext> {
+    private static final Logger LOG = Logger.getLogger(LocalHostOnlyFilter.class);
+
+    private static final String LOCAL_HOST = "localhost";
+    private static final String LOCAL_HOST_IP = "127.0.0.1";
+
+    public LocalHostOnlyFilter() {
+    }
+
+    @Override
+    public void handle(RoutingContext event) {
+        HttpServerResponse response = event.response();
+        if (hostIsValid(event)) {
+            event.next();
+        } else {
+            LOG.error("Dev UI: Only localhost is allowed");
+            response.setStatusCode(403);
+            response.setStatusMessage("Dev UI: Only localhost is allowed - Invalid host");
```

Comment:
I think there will be a problem with this.
Typically, the case I'm facing right now is that I need the dev UI to work from DevSpaces. Unfortunately, it's not localhost but an exposed route. 

We need to allow a list of allowed hosts.

Reply:
How about a config option where you can add allowed hosts ?

Reply:
You can now add hosts that should be allowed